edx_solutions_api_integration/courses/views.py

```diff
@@ -565,9 +567,14 @@ def get(self, request, course_id):
         """
         GET /api/courses/{course_id}
         """
+        try:
+            user = User.objects.get(username=request.GET.get('username'))
```

Comment:
If `username` is not passed, this will still call `User.objects.get(username=None)` which will do a pointless database query. Also, with this way of writing it, an invalid username would be ignored instead of returning an error - not what we want.

Instead, try something like:

```python
if 'username' in request.GET:
    try:
        user = User.objects.get(username=request.GET['username'])
    except User.DoesNotExist:
        raise Http404()
else:
    user = request.user  # Usually this is AnonymousUser since this is a server-to-server API
```